Java 12 functional sanity failure JCL_Test test_setSecurityManager2

I ran a Java 12 functional sanity test using the changes from #3980 and #4531.

https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/3
JCL_Test_0 JCL_Test_1
```
FAILED: test_setSecurityManager2
java.lang.AssertionError: Unexpected: java.lang.NullPointerException
	at org.testng.Assert.fail(Assert.java:96) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.openj9.test.java.lang.Test_System.test_setSecurityManager2(Test_System.java:363) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/functional/Java8andUp/GeneralTest.jar)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) from jrt:/java.base
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) from jrt:/java.base
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) from jrt:/java.base
	at java.base/java.lang.reflect.Method.invoke(Method.java:567) from jrt:/java.base
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:580) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:716) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:988) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestRunner.privateRun(TestRunner.java:648) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestRunner.run(TestRunner.java:505) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.runSuites(TestNG.java:1049) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.run(TestNG.java:1017) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.privateMain(TestNG.java:1354) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.main(TestNG.java:1323) from jdk.internal.loader.ClassLoaders$AppClassLoader@e635e82(file:/home/jenkins/jenkins-agent/workspace/Test-sanity.functional-JDK12-linux_390-64_cmprssptrs/jvmtest/TestConfig/lib/testng.jar)
```

--- a/jcl/src/java.base/share/classes/java/lang/System.java
+++ b/jcl/src/java.base/share/classes/java/lang/System.java
@@ -378,7 +378,10 @@ private static void ensureProperties() {
 	systemProperties.put("sun.jnu.encoding", platformEncoding); //$NON-NLS-1$
 	systemProperties.put("file.encoding", fileEncoding); //$NON-NLS-1$
 	systemProperties.put("file.encoding.pkg", "sun.io"); //$NON-NLS-1$ //$NON-NLS-2$
+	/*[IF !Java12]*/
+	/* System property java.specification.vendor is set via VersionProps.init(systemProperties) since JDK12 */
 	systemProperties.put("java.specification.vendor", "Oracle Corporation"); //$NON-NLS-1$ //$NON-NLS-2$
+	/*[ENDIF] !Java12 */
 	systemProperties.put("java.specification.name", "Java Platform API Specification"); //$NON-NLS-1$ //$NON-NLS-2$
 	systemProperties.put("com.ibm.oti.configuration", "scar"); //$NON-NLS-1$
 
@@ -397,6 +400,9 @@ private static void ensureProperties() {
 	/*[IF Java12]*/
 	/* java.lang.VersionProps.init() eventually calls into System.setProperty() where propertiesInitialized needs to be true */
 	java.lang.VersionProps.init(systemProperties);
+	/* VersionProps.init(systemProperties) above sets java.specification.version value which is used to set java.vm.specification.version. */
+	systemProperties.put("java.vm.specification.version", systemProperties.getProperty("java.specification.version")); //$NON-NLS-1$ //$NON-NLS-2$
+	systemProperties.put("java.vm.vendor", systemProperties.getProperty("java.vendor")); //$NON-NLS-1$ //$NON-NLS-2$
 	/*[ELSE]
 	/*[IF Sidecar19-SE]*/
 	/* java.lang.VersionProps.init() eventually calls into System.setProperty() where propertiesInitialized needs to be true */
@@ -507,12 +513,14 @@ static Properties internalGetProperties() {
  * The properties currently provided by the virtual
  * machine are:
  * <pre>
+/*[IF !Java12]
+ *     java.vendor
  *     java.vendor.url
+ /*[ENDIF] Java12
  *     java.class.path
  *     user.home
  *     java.class.version
  *     os.version
- *     java.vendor
  *     user.dir
  *     user.timezone
  *     path.separator

--- a/runtime/include/vendor_version.h
+++ b/runtime/include/vendor_version.h
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2017, 2018 IBM Corp. and others
+ * Copyright (c) 2017, 2019 IBM Corp. and others
  *
  * This program and the accompanying materials are made available under
  * the terms of the Eclipse Public License 2.0 which accompanies this
@@ -48,8 +48,16 @@
 
 #define VENDOR_SHORT_NAME "OpenJ9"
 
+#if JAVA_SPEC_VERSION < 12
+/* Pre-JDK12 version use following defines to set system properties
+ * java.vendor, java.vendor.url and java.vm.vendor within vmprop.c:initializeSystemProperties(vm).
+ * JDK12 (assuming future versions as well) sets these properties via java.lang.VersionProps.init(systemProperties) 
+ * and following settings within System.ensureProperties().
+ */
 #define JAVA_VENDOR "Eclipse OpenJ9"
 #define JAVA_VENDOR_URL "http://www.eclipse.org/openj9"
+#endif /* JAVA_SPEC_VERSION < 12 */
+
 #define JAVA_VM_NAME "Eclipse OpenJ9 VM"
 
 #endif     /* vendor_version_h */

--- a/runtime/vm/vmprops.c
+++ b/runtime/vm/vmprops.c
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 1991, 2018 IBM Corp. and others
+ * Copyright (c) 1991, 2019 IBM Corp. and others
  *
  * This program and the accompanying materials are made available under
  * the terms of the Eclipse Public License 2.0 which accompanies this
@@ -550,16 +550,13 @@ initializeSystemProperties(J9JavaVM * vm)
 	PORT_ACCESS_FROM_JAVAVM(vm);
 
 	J9VMSystemProperty *javaEndorsedDirsProperty;
-	jint i;
-	JavaVMInitArgs * initArgs;
+	jint i = 0;
+	JavaVMInitArgs * initArgs = NULL;
 	char * jclName = J9_DEFAULT_JCL_DLL_NAME;
-	J9JCLConfigurationInfo * jclConfig;
+	J9JCLConfigurationInfo * jclConfig = NULL;
 	UDATA j2seVersion = J2SE_VERSION(vm) & J2SE_VERSION_MASK;
 	const char* propValue = NULL;
-	const char* classVersion = NULL;
-	const char* specificationVersion = NULL;
-	const char* specificationVendor = NULL;
-	UDATA rc;
+	UDATA rc = J9SYSPROP_ERROR_NONE;
 
 	if (omrthread_monitor_init(&(vm->systemPropertiesMutex), 0) != 0) {
 		return J9SYSPROP_ERROR_OUT_OF_MEMORY;
@@ -594,40 +591,72 @@ initializeSystemProperties(J9JavaVM * vm)
 		}
 	}
 
-	/* Properties that always exist */
-	specificationVendor = "Oracle Corporation";
-	switch (j2seVersion) {
-		case J2SE_18:
-			classVersion = "52.0";
-			specificationVersion = "1.8";
-			break;
-
-		case J2SE_19:
-			classVersion = "53.0";
-			specificationVersion = "9";
-			break;
-			
-		case J2SE_V10:
-			classVersion = "54.0";
-			specificationVersion = "10";
-			break;
-			
-		case J2SE_V11:
-			classVersion = "55.0";
-			specificationVersion = "11";
-			break;
-
-		case J2SE_V12:
-			/* FALLTHROUGH */
-		default:
-			classVersion = "56.0";
-			specificationVersion = "12";
-			break;
+#if JAVA_SPEC_VERSION < 12
+	/* Following system properties are defined via java.lang.VersionProps.init(systemProperties) and following settings within System.ensureProperties() */
+	{
+		const char* classVersion = NULL;
+		const char* specificationVersion = NULL;
+		
+		/* Properties that always exist */
+		switch (j2seVersion) {
+			case J2SE_18:
+				classVersion = "52.0";
+				specificationVersion = "1.8";
+				break;
+	
+			case J2SE_19:
+				classVersion = "53.0";
+				specificationVersion = "9";
+				break;
+				
+			case J2SE_V10:
+				classVersion = "54.0";
+				specificationVersion = "10";
+				break;
+				
+			case J2SE_V11:
+				classVersion = "55.0";
+				specificationVersion = "11";
+				break;
+	
+			case J2SE_V12:
+				/* FALLTHROUGH */
+			default:
+				classVersion = "56.0";
+				specificationVersion = "12";
+				break;
+		}
+		rc = addSystemProperty(vm, "java.class.version", classVersion, 0);
+		if (J9SYSPROP_ERROR_NONE != rc) {
+			goto fail;
+		}
+	
+		rc = addSystemProperty(vm, "java.specification.version", specificationVersion, 0);
+		if (J9SYSPROP_ERROR_NONE != rc) {
+			goto fail;
+		}
+		
+		rc = addSystemProperty(vm, "java.vm.specification.version", specificationVersion, 0);
+		if (J9SYSPROP_ERROR_NONE != rc) {
+			goto fail;
+		}
+		
+		rc = addSystemProperty(vm, "java.vendor", JAVA_VENDOR, 0);
+		if (J9SYSPROP_ERROR_NONE != rc) {
+			goto fail;
+		}
+		
+		rc = addSystemProperty(vm, "java.vendor.url", JAVA_VENDOR_URL, 0);
+		if (J9SYSPROP_ERROR_NONE != rc) {
+			goto fail;
+		}
+		
+		rc = addSystemProperty(vm, "java.vm.vendor", JAVA_VENDOR, 0);
+		if (J9SYSPROP_ERROR_NONE != rc) {
+			goto fail;
+		}
 	}
-	rc = addSystemProperty(vm, "java.class.version", classVersion, 0);
-	if (J9SYSPROP_ERROR_NONE != rc) {
-		goto fail;
-	}
+#endif /* JAVA_SPEC_VERSION < 12 */
 
 	rc = addSystemProperty(vm, "com.ibm.oti.vm.library.version", J9_DLL_VERSION_STRING, 0);
 	if (J9SYSPROP_ERROR_NONE != rc) {
@@ -644,28 +673,7 @@ initializeSystemProperties(J9JavaVM * vm)
 		goto fail;
 	}
 
-	rc = addSystemProperty(vm, "java.vendor", JAVA_VENDOR, 0);
-	if (J9SYSPROP_ERROR_NONE != rc) {
-		goto fail;
-	}
-
-	rc = addSystemProperty(vm, "java.vendor.url", JAVA_VENDOR_URL, 0);
-
-	if (J9SYSPROP_ERROR_NONE != rc) {
-		goto fail;
-	}
-
-	rc = addSystemProperty(vm, "java.specification.version", specificationVersion, 0);
-	if (J9SYSPROP_ERROR_NONE != rc) {
-		goto fail;
-	}
-	
-	rc = addSystemProperty(vm, "java.vm.specification.version", specificationVersion, 0);
-	if (J9SYSPROP_ERROR_NONE != rc) {
-		goto fail;
-	}
-
-	rc = addSystemProperty(vm, "java.vm.specification.vendor", specificationVendor, 0);
+	rc = addSystemProperty(vm, "java.vm.specification.vendor", "Oracle Corporation", 0);
 	if (J9SYSPROP_ERROR_NONE != rc) {
 		goto fail;
 	}
@@ -676,11 +684,6 @@ initializeSystemProperties(J9JavaVM * vm)
 	}
 
 	rc = addSystemProperty(vm, "java.vm.version", J9JVM_VERSION_STRING, 0);
-	if (J9SYSPROP_ERROR_NONE != rc) {
-		goto fail;
-	}
-
-	rc = addSystemProperty(vm, "java.vm.vendor", JAVA_VENDOR, 0);
 	if (J9SYSPROP_ERROR_NONE != rc) {
 		goto fail;
 	}

